libs/ngx-mime/src/lib/viewer/viewer-footer/content-search-navigator/content-search-navigator.component.spec.ts

```diff
@@ -1,336 +1,154 @@
+  let viewerLayoutServiceSpy: Spy<ViewerLayoutService>;
+  let intl: MimeViewerIntl;
+  let loader: HarnessLoader;
+  let nextButton: MatButtonHarness;
+  let previousButton: MatButtonHarness;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
       schemas: [CUSTOM_ELEMENTS_SCHEMA],
       imports: [NoopAnimationsModule, SharedModule],
       declarations: [ContentSearchNavigatorComponent],
       providers: [
         MimeViewerIntl,
-        ContentSearchNavigationService,
-        { provide: ViewerService, useClass: ViewerServiceStub },
-        {
-          provide: IiifContentSearchService,
-          useClass: IiifContentSearchServiceStub,
-        },
-        { provide: CanvasService, useClass: CanvasServiceStub },
+        provideAutoSpy(IiifContentSearchService, {
+          observablePropsToSpyOn: ['onChange'],
+        }),
+        provideAutoSpy(ContentSearchNavigationService, {
+          observablePropsToSpyOn: ['currentHitCounter'],
+        }),
+        provideAutoSpy(CanvasService, {
+          observablePropsToSpyOn: ['onCanvasGroupIndexChange'],
+        }),
         { provide: IiifManifestService, useClass: IiifManifestServiceStub },
+        provideAutoSpy(ViewerLayoutService, {
+          gettersToSpyOn: ['layout'],
+        }),
       ],
     }).compileComponents();
-  }));
+  });
 
-  beforeEach(() => {
+  beforeEach(async () => {
     fixture = TestBed.createComponent(ContentSearchNavigatorComponent);
-    iiifContentSearchService = TestBed.inject<any>(IiifContentSearchService);
-    contentSearchNavigationService = TestBed.inject(
+    loader = TestbedHarnessEnvironment.loader(fixture);
+    iiifContentSearchServiceSpy = TestBed.inject(
+      IiifContentSearchService,
+    ) as Spy<IiifContentSearchService>;
+    intl = TestBed.inject(MimeViewerIntl);
+    contentSearchNavigationServiceSpy = TestBed.inject(
       ContentSearchNavigationService,
-    );
-    canvasService = TestBed.inject<any>(CanvasService);
+    ) as Spy<ContentSearchNavigationService>;
+    canvasServiceSpy = TestBed.inject(CanvasService) as Spy<CanvasService>;
+    viewerLayoutServiceSpy = TestBed.inject(
+      ViewerLayoutService,
+    ) as Spy<ViewerLayoutService>;
 
     component = fixture.componentInstance;
     component.searchResult = createDefaultData();
-    iiifContentSearchService._currentSearchResult.next(component.searchResult);
-    canvasService.addAll(createDefaultTileRects(102), ViewerLayout.TWO_PAGE);
+    iiifContentSearchServiceSpy.onChange.nextWith(component.searchResult);
     fixture.detectChanges();
+
+    nextButton = await loader.getHarness(
+      MatButtonHarness.with({
+        selector: '[data-testid="footerNavigateNextHitButton"]',
+      }),
+    );
+
+    previousButton = await loader.getHarness(
+      MatButtonHarness.with({
+        selector: '[data-testid="footerNavigatePreviousHitButton"]',
+      }),
+    );
```

Comment:
Should we move these out of the beforeEach and put them into own separate functions?